### PR TITLE
Dynamically select emulator ports from machine instead of static ports specified by user or defaults

### LINKED ISSUE #113: Emulator ports should not be static
### Instructions

Emulator ports should be autofilled based on available ports on system.  The end user should not have any context of what ports the emulator is running on as all of this information is unnecessary should be abstracted from their scope.

### Problem

The emulator frequently runs into conflicts with these static ports such that it attempts to open the same port as an already running emulator.

The user should not have to deal with these issues as they ultimately don't care which port the emulator is running on - they just want it to work

### PR DESCRIPTION
**Pull request by jribbink**
Closes #113

### CODE REVIEW

File: src/init.js [hunk 1]
```diff
@@ -27,7 +26,6 @@ import { config } from "@onflow/config";
  * @param {number} [props.pkey] - private key to use for service account in case of collisions
  */
 export const init = async (basePath, props = {}) => {
-  const { port } = props;
   const { pkey = "48a1f554aeebf6bf9fe0d7b5b79d080700b073ee77909973ea0b2f6fbc902" } = props;
```
Comment (gregsantos) [inline src/init.js:29]: What's this default key for?

**Review by gregsantos — CHANGES_REQUESTED** [review]
Should prob clean up the [README](https://github.com/onflow/flow-js-testing/tree/jribbink/auto-select-port#readme) and [docs/emulator.md](docs/emulator.md) reflect these changes and clarify how/if emulator should be started

File: src/init.js (hunk 1 above)
Comment (jribbink) [reply src/init.js:29]: Not sure, it was there before me... will look into whether that is necessary.

Comment (jribbink) [reply src/init.js:29]: I will make another PR for this as there is a bit more to unpack here than just this (docs, etc)

**Review by gregsantos — APPROVED** [review]
g2g

**Pull request merged**